tests/integration_tests/tests/agentless_tests/test_runtime_functions.py

```diff
@@ -0,0 +1,143 @@
+            node_id='node1', deployment_id=deployment.id)
+        self.assertEqual(len(instances_node1), 1)
+        instances_node2 = self.client.node_instances.list(
+            node_id='node2', deployment_id=deployment.id)
+        ni1 = instances_node1[0]
+        ni2 = instances_node2[0]
+
+        outputs = self.client.deployments.outputs.get(deployment.id)['outputs']
+        self.assertEqual(outputs['out1'], value)
+        self.assertEqual(outputs['out2'], value)
+        self.assertEqual(outputs['out3'], value)
+        self.assertEqual(ni1.runtime_properties['arg_value'], value)
+        self.assertEqual(ni1.runtime_properties['prop1_value'], value)
+        self.assertEqual(ni1.runtime_properties['prop2_value'], value)
+        self.assertEqual(ni1.runtime_properties['source_input_value'], value)
+        self.assertEqual(ni1.runtime_properties['target_input_value'], value)
+        self.assertEqual(ni2.runtime_properties['source_input_value'], value)
+        self.assertEqual(ni2.runtime_properties['target_input_value'], value)
+        self.assertEqual(ni1.runtime_properties['source_prop_value'], value)
+        self.assertEqual(ni2.runtime_properties['target_prop_value'], value)
+
+    def _manual_deployment_update(self, deployment):
+        update_script = resource('scripts/update_deployment.py')
+        self.copy_file_to_manager(update_script, '/tmp/update_deployment.py')
+
+        self.execute_on_manager(
+            '/opt/manager/env/bin/python /tmp/update_deployment.py '
+            '--deployment-id {0}'
+            .format(deployment.id))
+
+    def _update_deployment(self, deployment):
+        new_bp_path = os.path.join(
+            os.path.dirname(self.bp_path),
+            'altered_deployment_update_functions.yaml'
+        )
+        self.addCleanup(os.unlink, new_bp_path)
+        with open(self.bp_path) as bp, open(new_bp_path, 'w') as new_bp:
+            bp_content = bp.read()
+            bp_content = bp_content.replace(
+                "prop2: '{0}'".format(self.BASE_VALUE),
+                "prop2: '{0}'".format(self.CHANGED_VALUE))
+            new_bp.write(bp_content)
+
+        dep_update = self.client.deployment_updates.update(
+            deployment_id=deployment.id,
+            blueprint_or_archive_path=new_bp.name,
+            inputs={'input1': self.CHANGED_VALUE, 'fail_create': False}
+        )
+        execution = self.client.executions.get(dep_update.execution_id)
+        self.wait_for_execution_to_end(execution)
+        dep_update = self.client.deployment_updates.finalize_commit(
+            dep_update.id)
+
+    def test_manual_update_functions(self):
+        deployment = self.deploy(
+            self.bp_path,
+            inputs={'input1': self.BASE_VALUE},
+            functions_eager=False)
+        self.execute_workflow('install', deployment.id)
+        self._assert_properties(deployment, self.BASE_VALUE)
+        self._manual_deployment_update(deployment)
```

Comment:
why is this needed as an external script?

Reply:
because it must run on docl, because it uses storage-manager